fix: remove Node.js fs imports to enable React Native compatibility

Guid21 opened the pull request:
- Remove parseKeypairFile function from utils/signer.ts that was using require('fs')
- Modify client/services/userMetadata.ts to avoid fs.writeFileSync usage
- Keep noopSigner function which is required by main SDK classes
- Fixes React Native bundling error: 'attempted to import Node standard library module fs'

andreihrs commented on src/utils/userMetadata.ts:
```diff
@@ -230,28 +267,38 @@ function getLeverageObligationAndObligationFarmStateAddresses(
-  const obligationPdas: PublicKey[] = [];
-  const farmUserStates: PublicKey[] = [];
+): [{ address: PublicKey; log: string }[], { address: PublicKey; log: string }[]] {
+  const obligationPdas: { address: PublicKey; log: string }[] = [];
```
you can have a custom type for this since you reuse it a couple of times

timur-zakirov committed:
fix: remove Node.js fs imports to enable React Native compatibility

- Remove parseKeypairFile function from utils/signer.ts that was using require('fs')
- Modify client/services/userMetadata.ts to avoid fs.writeFileSync usage
- Keep noopSigner function which is required by main SDK classes
- Fixes React Native bundling error: 'attempted to import Node standard library module fs'
